File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/atendimento/atendimento/dto/AtendimentoGridDTO.java
package br.com.grupopipa.gestaointegrada.atendimento.atendimento.dto;

import java.time.LocalDateTime;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.core.dto.GridDTO;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Builder
@Getter
@Setter
public class AtendimentoGridDTO implements GridDTO {

    private UUID id;
    private Long numero;
    private LocalDateTime dataInicio;
    private String pacienteNome;
    private String profissionalAtendimentoNome;
    private int procedimentosCount;
    private String convenioNome;
    private LocalDateTime createdAt;
    private Boolean deleted;
}

File: src/frontend/src/app/components/atendimento/atendimento/model/atendimento-grid-dto.ts
import { Exclude, Expose } from 'class-transformer';

@Exclude()
export class AtendimentoGridDTO {
  @Expose() id?: string;
  @Expose() numero?: number;
  @Expose() dataInicio?: string;
  @Expose() pacienteNome?: string;
  @Expose() profissionalAtendimentoNome?: string;
  @Expose() procedimentosCount?: number;
  @Expose() convenioNome?: string;
  @Expose() createdAt?: Date;
  @Expose() deleted?: boolean;
}
